lib/Doctrine/DBAL/Connection.php

```diff
@@ -211,7 +218,7 @@ public function __construct(array $params, Driver $driver, Configuration $config
 
         if (isset($params["platform"])) {
             if ( ! $params['platform'] instanceof Platforms\AbstractPlatform) {
-                throw DBALException::invalidPlatformType($params['platform']);
+                throw InvalidPlatformType::new($params['platform']);
```

Comment:
Would it make sense to change the constructor signature and get rid of the exception?

Reply:
moved to #3163